docs/source/examples/maxcut-demo.myst

```diff
@@ -0,0 +1,481 @@
+
+    Returns:
+        A triple of the minimum cost, the optimal parameters and a list of costs at each iteration steps.
+    """
+   
+    
+    # store the optimization trajectory
+    traj = []
+
+    def callback(xk: np.ndarray) -> bool:
+        traj.append(cost_function(xk))
+        return True
+
+    results = minimize(
+        cost_function,
+        x0=init_params,
+        method="Nelder-Mead",
+        callback=callback,
+        options={"disp": True, "maxiter": 300},
+        # Set tol=0 to enforce a fixed number of iterations.
+        # Comment tol=0 to speedup the notebook execution.
+        tol=0,
+    )
+
+    plt.title("Optimization history")
+    plt.xlabel("Iteration Step")
+    plt.ylabel("Cost")
+    plt.plot(traj)
+    plt.show()
+    return results.fun, results.x, traj
+```
+
+## Comparing ideal, noisy and mitigated backends
+
++++
+
+### Ideal simulation
+
++++
+
+We focus on the square graph and try to solve the associated QAOA algorithm.
+First, we consider the ideal situation of a noiseless backend.
+
+```{code-cell} ipython3
+from matplotlib import pyplot as plt
+
+graph = [(0, 1), (1, 2), (2, 3), (3, 0)]
+betas = [0.0, 0.5]
+gammas = [0.7, 1.0]
+init_params = betas + gammas
+```
+
+```{code-cell} ipython3
+def qaoa_ideal_cost(params: np.array) -> float:
+    """Ideal cost function of the QAOA problem without noise."""
+    qaoa_circuit = qaoa_ansatz(graph, params)
+    qaoa_obs = qaoa_hamiltonian(graph)
+    return executor(qaoa_circuit, qaoa_obs, noise=0)
+
+ideal_opt_cost, ideal_opt_params, traj = minimize_cost(qaoa_ideal_cost, init_params)
+print("Ideal cost: ", np.round(ideal_opt_cost, 4))
+```
+
+Since the simulation is exact, we reach the optimal cost associated to (minus) the maximum number of cuts of the graph.
+
++++
+
+### Using a noisy unmitigated simulator
+
+We assume to have some depolarizing noise. This means that we need to re-define the QAOA cost function using a noisy executor function.
+
+```{code-cell} ipython3
+BASE_NOISE = 0.03
+def qaoa_noisy_cost(params: np.array) -> float:
+    """Noisy cost function of the QAOA problem."""
+    qaoa_circuit = qaoa_ansatz(graph, params)
+    qaoa_obs = qaoa_hamiltonian(graph)
+    return executor(qaoa_circuit, qaoa_obs, noise=BASE_NOISE)
+
+noisy_opt_cost, noisy_opt_params, noisy_traj = minimize_cost(qaoa_noisy_cost, init_params)
+```
+
+Because of noise, the QAOA optimization is not able to converge to the optimal solution. 
+
+Indeed the variational parameters optimized with noise do not minimize the ideal cost.
+
+```{code-cell} ipython3
+print("Noisy cost: ", np.round(noisy_opt_cost, 4))
+print("Ideal cost evaluated at noisy parameters: ", np.round(qaoa_ideal_cost(noisy_opt_params), 4))
+print("Ideal cost: ", np.round(ideal_opt_cost, 4))
+```
+
+### Using a noisy simulator mitigated with Mitiq
+
+Now let's try to mitigate this noise using Mitiq.
+
+This can be done by simply wrapping the noisy executor into a mitigated executor.
```

Comment:
```suggestion
This can be done by simply wrapping the noisy executor into a mitigated executor. We will use global folding and a linear inference (using a {ref}`Linear Factory <guide_zne_factory>` object) to implement zero-noise extrapolation.
```

Reply:
{side comment: The link is added in rst as the target file is in rst. This means that it does not show up an hyperlink if one is using Jupyter notebook to read the link, but it will correctly show up in the built documentation html file. In order to have the link work also when viewing the myst file with Jupyter, I think that the target file (the zne-guide) should be converted to myst.}